server/api/users/users.test.ts

```diff
@@ -37,19 +39,63 @@ describe('auth router', () => {
   });
 
   describe('GET /', () => {
+    it('API endpoint exists', async () => {
+      const res = await server.exec.get('/api/users/')
+      expect(res.status).toBe(200)
+    });
+
     it('expects empty array if no users exist', async () => {
       const res = await server.exec.get('/api/users/');
-      expect(res.body).toStrictEqual([]);
+      expect(res.body.page).toEqual(1)
+      expect(res.body.totalPages).toEqual(0)
+      expect(res.body.users).toStrictEqual([]);
+    });
+
+    it('expect invalid page request to return invalid page', async () => {
+        const res = await server.exec.get('/api/users/?page=-1');
+        expect(res.body.message).toEqual('Invalid page number');
+    });
+
+    it('expect invalid page request to return invalid page', async () => {
+        const res = await server.exec.get('/api/users/?page=a');
+        expect(res.body.message).toEqual('Invalid page number');
+    });
+
+    it('expect page request beyond page limit to be out of range', async () => {
+        const res = await server.exec.get('/api/users/?page=100');
+        expect(res.body.message).toEqual('Page out of range');
     });
 
-    it('returns all users', async () => {
-      const userOne = await createUser();
-      const userTwo = await createUser();
+    it('returns an array of 20 users', async () => {
+      const count = 25
+      const users: User[] = await Promise.all(Array.from({ length: count }, createUser));
+      users.sort((a,b) => a.id - b.id)
+
+      const limit = 20;
+      const totalPages = Math.ceil(count / limit);
 
       const res = await server.exec.get('/api/users/');
       expect(res.status).toBe(200);
-      expect(res.body[0]).toEqual(getExpectedUserObject(userOne));
-      expect(res.body[1]).toEqual(getExpectedUserObject(userTwo));
+      expect(res.body.users.length).toEqual(limit);
+      expect(res.body.page).toEqual(1);
+      expect(res.body.totalPages).toEqual(totalPages);
+    });
+
+    it('returns last 5 users when it hits the second page', async () => {
+      const count = 25
+      const users: User[] = await Promise.all(Array.from({ length: count }, createUser));
+      users.sort((a,b) => a.id - b.id)
+
+      const limit = 20;
+      const totalPages = Math.ceil(count / limit);
+
+      const res = await server.exec.get('/api/users?page=2');
+
+      expect(res.status).toBe(200);
+      expect(res.body.users.length).toEqual(5);
+      expect(res.body.users[0]).toEqual(getExpectedUserObject(users[20]));
```

Comment:
@timmyichen @devonzara 

This is occasionally failing. I believe it has to do with the Promise.all not guaranteeing ids are put into the array in sequential order. I've attempted a `users.sort((a,b) => a.id - b.id)` but that has not fixed the anomaly.